### ISSUE TITLE
Heat scale color range should be wider for better accessibility

### ISSUE THREAD

**Issue by anne-stevens** [carbon-design-system/carbon-charts#503]
#### I am submitting a...

- [ ] Feature request
- [x] Design defect
- [ ] Source code defect
- [x] Demo/documentation defect
- [ ] Other

#### charts version:
This page on Carbon Design System -> Data visualization -> Color palettes page: https://www.carbondesignsystem.com/data-visualization/color-palettes

### Issue description
This issue was discussed in a meeting with @sadek and @mjabbink

#### Steps to produce the issue
Go to this page on Carbon Design System -> Data visualization -> Color palettes page: https://www.carbondesignsystem.com/data-visualization/color-palettes

#### Current behavior
Existing documentation about purple heat scale (blue is similar)
![Screen Shot 2020-02-18 at 4 00 41 PM](https://user-images.githubusercontent.com/59665153/74777726-132cc900-5268-11ea-9742-c46c5ffd3db3.png)

The way the heat scale palettes are presented in the Carbon Charts page suggests that on a light background, the colours range from purple 50 to purple 100 and on a dark background the colours range from purple 60 to purple 10.  As a result, the data range is mapped to a narrower colour range than necessary and it actually makes the colour in the viz less accessible than necessary.

Here is an example of the results:
![Screen Shot 2020-02-18 at 3 14 43 PM](https://user-images.githubusercontent.com/59665153/74777908-7a4a7d80-5268-11ea-8cc2-afff94900150.png)

#### Expected behavior
I expect to be able to map the data to a much wider colour range (eg. Purple 10 to purple 90), similar to below (only not blue).

![Screen Shot 2020-02-18 at 3 14 53 PM](https://user-images.githubusercontent.com/59665153/74778030-c1387300-5268-11ea-9b0e-48a483e6bb81.png)

Widening the colour range makes it easier to distinguish shades of purple within the viz and improves accessibility.

**Comment by theiliad** [comment]
Related to https://github.com/carbon-design-system/carbon-website/issues/896

**Comment by mjabbink** [comment]
@anne-stevens that is not the correct guidance for heat maps. What you are describing is not the guidance and if that is specified somewhere it’s wrong and needs to be corrected

**Comment by mjabbink** [comment]
What you are showing as carbon guidance is not the guidance

**Comment by mjabbink** [comment]
@anne-stevens 
https://www.carbondesignsystem.com/data-visualization/advanced-charts#heatmap

https://www.carbondesignsystem.com/data-visualization/color-palettes#diverging-palettes

the 2-color diverging samples are currently limited so we can certainly extend them to include more options. This color page is also a serious work in progress and needs updating and design improvements.

**Comment by mjabbink** [comment]
carbon heatmap sample:

![charts-adv-heatmap](https://user-images.githubusercontent.com/32881239/74802726-b3a2dd80-52a8-11ea-9a40-2680929bfad3.png)

**Comment by mjabbink** [comment]
<img width="684" alt="Screen Shot 2020-02-18 at 11 45 55 PM" src="https://user-images.githubusercontent.com/32881239/74802788-ddf49b00-52a8-11ea-849a-581ca404eefe.png">


—
<img width="1068" alt="Screen Shot 2020-02-18 at 11 16 54 PM" src="https://user-images.githubusercontent.com/32881239/74802820-ebaa2080-52a8-11ea-90ff-7e3a9efd660c.png">

<img width="1011" alt="Screen Shot 2020-02-18 at 11 17 06 PM" src="https://user-images.githubusercontent.com/32881239/74802817-eb118a00-52a8-11ea-9568-b500e14ade19.png">

**Comment by mjabbink** [comment]
@theiliad the guidance here has been misinterpreted. I’ll cc @shixiedesign @cameroncalder so they are in the loop on this.

**Comment by theiliad** [comment]
I agree, I think there are multiple mistakes within the carbon website guidelines. @jeanservaas has caught most of those and is looking to fix soon I believe

**Comment by theiliad** [comment]
> @theiliad the guidance here has been misinterpreted. I’ll cc @shixiedesign @cameroncalder so they are in the loop on this.

I agree, I think there are multiple mistakes within the carbon website guidelines. @jeanservaas has caught most of those and is looking to fix soon I believe

**Comment by anne-stevens** [comment]
In a case such as this, below ...
![Screen Shot 2020-02-19 at 1 28 25 PM](https://user-images.githubusercontent.com/59665153/74863397-d28e8780-531b-11ea-98db-5d189b35d300.png)

... where there is no inflection point in the data (eg. a 0 value midway between -100 and +100) a one directional, sequential heat scale such as light purple to dark purple would be better viz practice.  And this is an example of where a wider heat scale would improve accessibility.  

Happy to discuss typical conditions to address in the guidance pages, if that would be helpful.

**Comment by mjabbink** [comment]
@anne-stevens we would love it if you could help advise Jeannie and I to get our guidance more clear and communicating best practices. We also have Accurat to help improve and update the pages on Carbon. When we connect to go over the pages and what we already have captured to resolve I will invite you to also participate.

**Comment by mjabbink** [comment]
@anne-stevens The range you’re looking for is available in the color palette so again maybe our guidance is not clear. but you have 10—100 in any color.

**Comment by mjabbink** [comment]
the brighter example includes the range. B10–B100 and R10—R100. https://github.com/carbon-design-system/carbon-charts/issues/503#issuecomment-588033396

**Comment by mjabbink** [comment]
but to Sadek’s point in the Slack channel getting into the lighter colors is not great for accessibility and would depend on some other interactive feature to carry communication. Some of the accessibility requirements not clear to me in some data viz so getting Bo Campbell or some from the accessibility team to weight in on this would be helpful

**Comment by theiliad** [comment]
@mjabbink 
within the heatmap component specifically I'd also suggest adding a 1-2px gap between each rectangle, that should guarantee accessibility for the most part

**Comment by theiliad** [comment]
Example
![image](https://user-images.githubusercontent.com/14989804/74957971-47be9300-53d6-11ea-8a87-93b3bc60ba1f.png)

**Comment by anne-stevens** [comment]
I know this doesn't make the accessibility problem go away, but there is an exception in the accessibility standards for heat maps.  The 3:1 min contrast requirement does not apply.  I've confirmed w/ Michael Gower. See:
https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

Also, the tooltip would provide detailed information.

**Comment by theiliad** [comment]
> I know this doesn't make the accessibility problem go away, but there is an exception in the accessibility standards for heat maps. The 3:1 min contrast requirement does not apply. I've confirmed w/ Michael Gower. See:
> https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html
> 
> Also, the tooltip would provide detailed information.

Yes that's mainly been my concern with going lower than shades of 50, we might not necessarily need to do that specially because some of the very light shades can cause problems on their own.

I think the spacing between rectangles can add a massive amount of value here

**Comment by mjabbink** [comment]
@anne-stevens that is great and confirms we can use the full 10–100 ranges. @theiliad I think it would be nice to offer both. In larger more simple het maps the line may not be needed where as in the sample you show above it may be very helpful. Both are certainly within the IDL in terms of style and aesthetic.

**Comment by theiliad** [comment]
> @anne-stevens that is great and confirms we can use the full 10–100 ranges. @theiliad I think it would be nice to offer both. In larger more simple het maps the line may not be needed where as in the sample you show above it may be very helpful. Both are certainly within the IDL in terms of style and aesthetic.

👍

**Comment by theiliad** [comment]
@mjabbink @jeanservaas is this something you'd be able to revise?

**Comment by mjabbink** [comment]
We will revise and add to the long list of others items to revise in the website.
